exercises/concept/role-playing-game/.docs/introduction.md

```diff
@@ -1,2 +1,81 @@
 # Introduction
 
+## Null-References
+
+If you have ever used another programming language (C/C++, Python, Java, Ruby, Lisp, etc.), it is likely that you have encountered `null` or `nil` before. 
+The use of `null` or `nil` is the way that these langugages indicate that a particular variable has no value. 
+However, this makes accidentally calling a variable that points to `null` an easy (and frequent) mistake to make. 
+As you might imagine, trying to call a function that isn't there, or access a value that doesn't exist can lead to all sorts of bugs and crashes. 
+The creator of `null` went so far as to call it his ['billion-dollar mistake'.](https://www.infoq.com/presentations/Null-References-The-Billion-Dollar-Mistake-Tony-Hoare/)
+
+## The Option Type
+
+To avoid these problems, Rust does not use null-references.
+However, it still needs a (safe) way to indicate that a particular variable has no value.
+This is where `Option` comes in.
+Instead of having a variable referencing `null` directly, Rust variables can point to the special `Option` enum.
+This enum, in turn, contains two possible variants: `None` - Rust's null-equivalent; and `Some(T)` - where T is any value of any type.
```

Comment:
```suggestion
Instead of having a variable which lacks a value, Rust variables can use the `Option` enum.
This enum has two variants: `None`, Rust's null-equivalent; and `Some(T)`, where T is a value of any type.
```

The whole point of `Option` is that it is _not_ special; it's just a normal enum, with some handy methods and traits defined on it. You could define your own version of `Option` with its full feature set, if you really wanted to.

The thing which made null pointers into a mistake, instead of a feature, is that they are special: they're a compiler hack, a value with a special meaning. The fix is in making them into a normal instance of a normal type.